File: src/main/java/aurocosh/divinefavor/common/item/talismans/spell/SpellTalismanFlood.java
package aurocosh.divinefavor.common.item.talismans.spell;

import aurocosh.divinefavor.common.config.common.ConfigSpells;
import aurocosh.divinefavor.common.item.talismans.spell.base.ItemSpellTalisman;
import aurocosh.divinefavor.common.item.talismans.spell.base.SpellOptions;
import aurocosh.divinefavor.common.item.talismans.spell.base.TalismanContext;
import aurocosh.divinefavor.common.lib.wrapper.MaterialPredicate;
import aurocosh.divinefavor.common.lib.wrapper.PredicateWrapper;
import aurocosh.divinefavor.common.spirit.base.ModSpirit;
import aurocosh.divinefavor.common.util.UtilBlock;
import aurocosh.divinefavor.common.util.UtilCoordinates;
import aurocosh.divinefavor.common.util.UtilList;
import net.minecraft.block.BlockLiquid;
import net.minecraft.block.material.Material;
import net.minecraft.block.state.IBlockState;
import net.minecraft.init.Blocks;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.World;

import java.util.EnumSet;
import java.util.List;

public class SpellTalismanFlood extends ItemSpellTalisman {
    public SpellTalismanFlood(String name, ModSpirit spirit, int favorCost, EnumSet<SpellOptions> options) {
        super(name, spirit, favorCost, options);
    }

    @Override
    protected void performActionServer(TalismanContext context) {
        World world = context.world;
        List<BlockPos> posList = UtilCoordinates.INSTANCE.getBlocksInSphere(context.pos, ConfigSpells.flood.radius);
        posList = UtilList.select(posList, new MaterialPredicate(world, Material.WATER));
        posList = UtilList.select(posList, new PredicateWrapper<>(world::getBlockState, state -> state.getValue(BlockLiquid.LEVEL) != 0));

        IBlockState state = Blocks.WATER.getDefaultState();
        for (BlockPos pos : posList)
            UtilBlock.replaceBlock(context.player, world, pos, state);
    }
}

File: src/main/java/aurocosh/divinefavor/common/item/talismans/spell/SpellTalismanFlood.kt
package aurocosh.divinefavor.common.item.talismans.spell

import aurocosh.divinefavor.common.config.common.ConfigSpells
import aurocosh.divinefavor.common.item.talismans.spell.base.ItemSpellTalisman
import aurocosh.divinefavor.common.item.talismans.spell.base.SpellOptions
import aurocosh.divinefavor.common.item.talismans.spell.base.TalismanContext
import aurocosh.divinefavor.common.lib.extensions.getMaterial
import aurocosh.divinefavor.common.spirit.base.ModSpirit
import aurocosh.divinefavor.common.util.UtilBlock
import aurocosh.divinefavor.common.util.UtilCoordinates
import aurocosh.divinefavor.common.lib.extensions.filter
import net.minecraft.block.BlockLiquid
import net.minecraft.block.material.Material
import net.minecraft.init.Blocks
import java.util.*

class SpellTalismanFlood(name: String, spirit: ModSpirit, favorCost: Int, options: EnumSet<SpellOptions>) : ItemSpellTalisman(name, spirit, favorCost, options) {

    override fun performActionServer(context: TalismanContext) {
        val world = context.world
        val posList = UtilCoordinates.getBlocksInSphere(context.pos, ConfigSpells.flood.radius)
        val flowingWaterBlocks = posList
                .filter(world::getMaterial, Material.WATER::equals)
                .filter(world::getBlockState) { state -> state.getValue(BlockLiquid.LEVEL) !== 0 }

        val state = Blocks.WATER.defaultState
        for (pos in flowingWaterBlocks)
            UtilBlock.replaceBlock(context.player, world, pos, state)
    }
}
